code/modules/power/battery.dm

```diff
@@ -297,23 +298,10 @@ var/global/list/battery_online =	list(
 
 
 /obj/machinery/power/battery/emp_act(severity)
-
-	var/old_online = online
-	var/old_charging = charging
-	var/old_output = output
-
-	online = 0
-	charging = 0
-	output = 0
 	charge = max(0, charge - 1e6/severity)
-
+	stat |= FORCEDISABLE
```

Comment:
There's already a flag for being emp'd, which is what should be used here. I recommend FORCEDISABLE is reserved for events to make it clear what each flag is for.